dnf/cli/commands/check.py

```diff
@@ -51,7 +51,8 @@ def set_argparser(parser):
                             action='append_const', const='provides',
                             help=_('show problems with provides'))
         # Add compatibility with yum but invisible in help
-        parser.add_argument('check_yum_types', nargs='*',
+        parser.add_argument('check_yum_types', nargs='*', choices=[
+            'all', 'dependencies', 'duplicates', 'obsoleted', 'provides', []],
```

Comment:
@mluscon @ignatenkobrain The argument is 'check_yum_types' is taken from yum, but dnf replace it with options --all, --dependecies and so on. The argument is able to handle certain key words (like in yum), but if unknown keyword is used, it doesn't complain and return no problem (like in yum). The patch should inform user, that used unrecognized argument. The argument is optional, but the combination with choices makes trouble, because it try in case of absence return [] but it is unsupported value, therefore add [] to choices is solution, how to handle the problem. The option is hidden in help, only appears error with list of choices if unsupported value appears.